docs/v2_v3_migration.md

```diff
@@ -11,58 +11,93 @@ Migration from v2 to v3 Helm
+       - install: requires release name or `--generate-name` argument
+       - inspect --> show
+       - registry: login to or logout from a registry
+         - login: login to a registry
+         - logout: logout from a registry
+       - reset (removed)
+       - serve (removed)
+       - upgrade: Added argument `maxHistory` which limits the maximum number of revisions saved per release (0 for no limit)
    - Improvements to error responses
    - Improvements on command argument validation and argument naming
    - Dynamically linked chart deopendencies moved to `Chart.yaml` (`requirements.yaml` removed and  requirements --> dependencies)
    - Library charts (helper charts) can now be added as dynamically linked chart dependencies
+   - Charts have a `type` metadata field to define the chart to be of an `application` or `library` chart. It is application by
+     default which means it is renderable and installable.
+   - Go import path is changed from `k8s.io/helm` to `helm.sh/helm` 
 
 ## Migration Use Cases
 
 The migration use cases are as follows:
 
 1. Running Helm v2 and v3 concurrently on the same cluster:
-   - v2 and v3 history/state are independent of each other. v2 uses "Configmaps" under Tiller namespace and ownership. v3 uses "Secrets" in user namespace and Helm ownership. There should be no conflicts.
-   - The only issue could be if Kubernetes resources are not named with unique capability like with rease in its name and you depoy the chart again for v3. This would happen for v2 anyway when not an upgrade. Can be avoided by naming resources uniqyely.
-   - Make sure to not override the v2 binary (`helm`)
-   - TBC: Will v3 have configuration and if so will it be the same directory as v2?
+   - v2 and v3 history/state are independent of each other. v2 uses "ConfigMaps" under Tiller namespace and `TILLER`ownership. v3 uses "Secrets" in user namespace and `helm` ownership. There should be no conflicts.
+   - The only issue could be if Kubernetes resources are not named with unique capability like with release in its name and you depoy the chart again for v3. This would happen for v2 anyway when not an upgrade. Can be avoided by naming resources uniquely.
+   - Make sure to not override the v2 client binary (`helm`). Rename or separate directory.
+   - v3 has no configuration and therefore doesn't need to be initialized. Run as is.
  
 2. Deploying a new chart:
    - Use v3 to deploy
  
-3. Upgrading an existing release:
-   - Deploy as new on v3 and delete the existing release on v2
+3. Upgrading an existing release from v2 to v3:
+   - The choices are as follows:
+     - Lose the release history: Deploy as new using v3 and delete the existing release using v2
+     - Maintain the release history: TBD: Detail the steps involved in migration of v2 history to v3. This should describe the steps manually and then any tools provided for those manual steps. Are the steps involved:
+       - Retrieve ConfigMaps for Tiller owner
```

Comment:
v2 releases could also be stored in secrets

Reply:
Good point, thanks

Reply:
Sure. Updated